xuluomin/src/server/src/server.cpp

```diff
@@ -0,0 +1,77 @@
+#include <ros/ros.h>
+#include <time.h>
+
+#include <server/time.h>
+#include <server/show.h>
+
+#include <vector>
+#include <iostream>
+
+server::show srv;
+
+ros::NodeHandle *n_pointer;
+
+std::vector<ros::Subscriber> server_subscribers;
+
+ros::ServiceServer service_server;
+
+void time_callback(const server::time::ConstPtr& time)
+{
+    time_t  cur_t=time->second;
+    struct tm *t=localtime(&cur_t);
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Consider using localtime_r(...) instead of localtime(...) for improved thread safety.  [runtime/threadsafe_fn] [2]